rust/agents/kathy/src/kathy.rs

```diff
@@ -44,6 +44,18 @@ impl Agent for Kathy {
                 .await?,
         ))
     }
+
+    #[allow(clippy::async_yields_async)]
```

Comment:
I take it this is just to allow the fn to be implemented even if there isn't necessarily any async stuff happening in it?

Reply:
No, this is to allow an async function to return somethning which is also a Future. Basically having `Future<Output=Future<Output=T>>` is generally bad, however here is special because the innermost future is actually a wrapped handle.